rotkehlchen/accounting/events.py

```diff
@@ -458,6 +458,19 @@ def add_sell_and_corresponding_buy(
         - RemoteError if there is a problem reaching the price oracle server
         or with reading the response returned by the server
         """
+        # In the case where the fee asset is nominated in the asset that we receive
+        # take care that the fee is processed once we have the received asset in possesion.
+        if selling_asset == fee_currency:
```

Comment:
Again here the comment says a different thing than what the code does.

Comment: 
> In the case where the fee asset is nominated in the asset that we receive

Code: 
`if selling_asset == fee_currency`

For them to be equal it should be:
`if receiving_asset == fee_currency:`

Also reading the if/else I am confused.

If selling_asset == fee_currency -> the same thing as before happens
otherwise -> you move the fee asset/amount to the buy instead of the sell.


What I assume you wanted to do and would also match the comment would be:

```python
if receiving_asset == fee_currency:
     buy_fee_asset = fee_currency
     buy_fee_amount = fee_amount
     sell_fee_asset = None
     sell_fee_amount = Fee(ZERO)
else:
      buy_fee_asset = None
      buy_fee_amount: Optional[Fee] = Fee(ZERO)
      sell_fee_asset = fee_currency
      sell_fee_amount = fee_amount
```